### PR TITLE
Mbliakharskyi/asana

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/asana/src/inngest/client.ts [hunk 1]
```diff
@@ -13,6 +13,12 @@ export const inngest = new Inngest({
         offset?: string;
       };
     };
+    'token/refresh': {
+      data: {
+        organisationId: string;
+        refreshTokenInfo: string;
```
Comment (elbalexandre) [inline apps/asana/src/inngest/client.ts:19]:
```suggestion
        refreshToken: string;
```
You can replace `refreshTokenInfo` by `refreshToken` everywhere.

**Review by elbalexandre — COMMENTED** [review]
Nice work,
(don't forget to remove the webhook route & the webhook connector)
Could you run theses two command locally:
`npx next lint --fix`
`npx prettier . --write`

in your asana directory. For next integration it could be easier to have vscode extension for prettier & eslint installed.

File: apps/asana/src/inngest/client.ts (hunk 1 above)
Comment (mbliakharskyi) [reply apps/asana/src/inngest/client.ts:19]: import { refreshToken, type RefreshTokenResponseData } from '@/connectors/auth';

As you can see, I already created the same named function. It conflicts.

File: apps/asana/src/inngest/functions/users/schedule-token-refresh.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { inngest } from '../../client';
+
+export const scheduleTokenRefresh = inngest.createFunction(
+  { id: 'schedule-token-refresh' },
+  { cron: env.TOKEN_REFRESH_CRON },
+  async ({ step }) => {
+<<<<<<< HEAD
+=======
+
+>>>>>>> c744f00a505206d20e48a9d0acead4cdce1a75e5
```
Comment (elbalexandre) [inline apps/asana/src/inngest/functions/users/schedule-token-refresh.ts:14]:
```suggestion
```

File: apps/asana/src/database/schema.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export const Organisation = pgTable('organisation', {
   email: text('email').notNull(),
   accessToken: text('access_token').notNull(),
   refreshToken: text('refresh_token').notNull(),
-  expiresIn: bigint('expires_in', { mode: 'number' }).notNull(),
+  expiresAt: timestamp('expires_at').notNull(),
   asanaId: text('asana_id').notNull(),
   gid: text('gid').notNull(),
   webhookSecret: text('webhook_secret'),
```
Comment (elbalexandre) [inline apps/asana/src/database/schema.ts:13]:
```suggestion
```
Theses fields are unused, we can remove them.

you will have to regenerate the `/drizzle` folder.

File: apps/asana/src/database/schema.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const Organisation = pgTable('organisation', {
   email: text('email').notNull(),
```
Comment (elbalexandre) [inline apps/asana/src/database/schema.ts:7]: We are not using neither the name or the email.
```suggestion
```

File: apps/asana/src/connectors/webhook.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ export const registerWebhook = async (organisationId: string, accessToken: strin
     Asana.ApiClient.instance.authentications.token = accessToken;
```
Comment (elbalexandre) [inline apps/asana/src/connectors/webhook.ts:9]: This file can be removed

File: apps/asana/src/inngest/client.ts (hunk 1 above)
Comment (elbalexandre) [reply apps/asana/src/inngest/client.ts:19]: Ok, no problem.

File: apps/asana/.env.test [hunk 6]
```diff
@@ -1,17 +0,0 @@
-ELBA_SOURCE_ID="0442f541-45d2-487a-9e4b-de03ce4c559e"
```
Comment (elbalexandre) [inline apps/asana/.env.test:1]: Can you re-add this file please. It will be needed for test pipeline.

Comment (mbliakharskyi) [reply apps/asana/.env.test:1]: Okay

**Pull request merged**